docs/additional-examples/evm-swap.md

```diff
@@ -0,0 +1,658 @@
+import { ethers } from "ethers";
+import bs58 from "bs58";
+```
+
+### Creating our two wallet instances
+
+These wallets will represent the two parties involved in the swap.
+
+```js
+async function getWalletA() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_1,
+        provider
+    );
+    return wallet;
+}
+
+async function getWalletB() {
+    const provider = new ethers.providers.JsonRpcProvider(
+        `https://yellowstone-rpc.litprotocol.com/`
+    );
+    const wallet = new ethers.Wallet(
+        process.env.NEXT_PUBLIC_PRIVATE_KEY_2,
+        provider
+    );
+    return wallet;
+}
+```
+
+### Creating our Lit Action and uploading it to IPFS
+
+Once we are done with our Lit Action, we need to upload it to IPFS so we get an immutable CID which will always point to our action.
+
+```js
+export async function createLitAction() {
+    console.log("creating lit action..");
+    const action = createERC20SwapLitAction(chainAParams, chainBParams);
+    const ipfsCid = await uploadViaPinata(action);
+
+    console.log("Lit Action code:\n", action);
+    console.log("IPFS CID: ", ipfsCid);
+    return ipfsCid;
+}
+```
+
+### Create a Mint/Grant/Burn PKP
+
+:::info
+Currently, direct calls to the smart contract’s m/g/b function are currently failing with the Lit Network. We’ll replicate the same functionality using the SDK.
+:::
+
+We'll be using `mintNextAndAddAuthMethods` to perform 3 calls in a single transaction:
+
+- Mint a PKP with the user's wallet
+- Add the Lit Action as a permitted auth method
+- 'Burn' the PKP NFT by transferring it to itself so that no additional auth methods can be added. Effectively, this offers the same functionality as the M/G/B method
+
+Now our PKP can only be used to execute the Lit Action we permitted when we added it as an auth method. It will only generate signatures when the Lit Action is running and the conditions we set have been met.
+
+```js
+export async function mintGrantBurnPKP(action_ipfs, mintedPKP) {
+    console.log("minting started..");
+    const signerA = await getWalletA();
+
+    const litContracts = new LitContracts({
+        signer: signerA,
+        network: LitNetwork.DatilDev,
+        debug: false,
+    });
+
+    const bytesAction = await stringToBytes(action_ipfs);
+    const pkpMintCost = await litContracts.pkpNftContract.read.mintCost();
+
+    await litContracts.connect();
+
+    const tx =
+        await litContracts.pkpHelperContract.write.mintNextAndAddAuthMethods(
+            AuthMethodType.LitAction,
+            [AuthMethodType.LitAction],
+            [bytesAction],
+            ["0x"],
+            [[AuthMethodScope.SignAnything]],
+            false,
+            true,
+            {
+                value: pkpMintCost,
+                maxFeePerGas: ethers.BigNumber.from("1800000000"),
```

Comment:
I don't think this property is required?

Reply:
my transaction was failing so I had to configure the gases manually